pkg/cas/content_addressable_storage_server.go

```diff
@@ -47,7 +47,36 @@ func (s *contentAddressableStorageServer) FindMissingBlobs(ctx context.Context,
 }
 
 func (s *contentAddressableStorageServer) BatchReadBlobs(ctx context.Context, in *remoteexecution.BatchReadBlobsRequest) (*remoteexecution.BatchReadBlobsResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "This service does not support batched reading of blobs")
+	// Asynchronously call Get() for every blob.
+	responsesChan := make(chan *remoteexecution.BatchReadBlobsResponse_Response, len(in.Digests))
+	for _, reqDigest := range in.Digests {
+		go func(reqDigest *remoteexecution.Digest) {
+			digest, err := digest.NewDigestFromPartialDigest(in.InstanceName, reqDigest)
+			var buf buffer.Buffer
+			if err == nil {
+				buf = s.contentAddressableStorage.Get(
+					ctx,
+					digest)
+			}
+			maxSizeBytes, err := buf.GetSizeBytes()
```

Comment:
This calls `GetSizeBytes()` on a nil buffer in case the branch above is not taken.

Also keep in mind that `GetSizeBytes()` is only of use in code that is CAS/AC agnostic. For blobs coming from the CAS, you can simply use `digest.GetSizeBytes()`.